src/networking.c

```diff
@@ -1332,9 +1333,8 @@ void unlinkClient(client *c) {
 
     /* Remove from the list of pending writes if needed. */
     if (c->flags & CLIENT_PENDING_WRITE) {
-        ln = listSearchKey(server.clients_pending_write,c);
-        serverAssert(ln != NULL);
-        listDelNode(server.clients_pending_write,ln);
+        serverAssert(&c->clients_pending_write_node != NULL);
```

Comment:
This assert is always true, i think you meant to check prev and next

Reply:
I don't think it makes sense to assert that prev and next are null here,
maybe we can just get rid of this assert?
I had it removed in the first revision but added it back at Maddy suggestion.

We can assert that prev and next are null after unlinking... but i don't think that is the idea the original assert had in mind

Reply:
The original point of the assert, imo, was that there is an independent flag that infers whether or not this value is correct. I was expecting next and prev (or just one of the two really). I don't feel strongly either way.

Reply:
I'll assert that one of the two is not null